src/som/compiler/MixinBuilder.java

```diff
@@ -385,10 +382,11 @@ public MixinDefinition assemble(final SourceSection source) {
   }
 
   private boolean outerScopeIsImmutable() {
-    if (outerBuilder == null) {
+    if (getFirstOuterMixin() == null) {
```

Comment:
should that be `isModule()`? that would an unrelated change improvement, but I noticed, because there you used `getContext()` instead of `getFirstOuterMixin()`.